src/types/tdigest.h

```diff
@@ -150,3 +151,74 @@ inline StatusOr<double> TDigestQuantile(TD&& td, double q) {
   diff /= (lc.weight / 2 + rc.weight / 2);
   return Lerp(lc.mean, rc.mean, diff);
 }
+
+template <typename TD>
+inline Status TDigestRank(TD&& td, const std::vector<double>& inputs, std::vector<int>& result) {
```

Comment:
This function seems specialized for `RevRank` rather than `Rank`.

Could we implement the `Rank`, then wrap the `Iterator` to a reverse version with the same logic to construct `RevRank`?